framework/update/helper/updateui_win.cpp

```
@@ -0,0 +1,642 @@
+{
+    LOGFONTW logFont = { };
+    logFont.lfHeight = -::MulDiv(MESSAGE_POINT_SIZE, static_cast<int>(dpi), 72);
+    logFont.lfWeight = FW_NORMAL;
+    logFont.lfCharSet = DEFAULT_CHARSET;
+    logFont.lfQuality = CLEARTYPE_QUALITY;
+    ::wcscpy_s(logFont.lfFaceName, L"Segoe UI");
+
+    return ::CreateFontIndirectW(&logFont);
+}
+
+UINT windowDpi(HWND hwnd)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)